### PR TITLE
Feature/28858 libreoffice saga continues

### PR DESCRIPTION
**Pull request by dals83**

### CODE REVIEW

File: scrapy-webscanner/process_queue.py [hunk 1]
```diff
@@ -20,9 +20,19 @@
...
 """
 
+import signal
+
+def sigterm_handler(sig, frm):
+    sys.exit(1)
+
+# process needs to listen on
+# signal.SIGTERM in order to tear down ressources.
+signal.signal(signal.SIGTERM, sigterm_handler)
```
Comment (danchr) [inline scrapy-webscanner/process_queue.py]: Move this down a bit; code should not precede imports unless strictly necessary.

File: scrapy-webscanner/process_manager.py [hunk 2]
```diff
@@ -177,6 +177,106 @@ def main():
...
+        restart_stuck_processors()
+
+        check_running_scanjobs()
+
+        # Cleanup finished scans from the last minute
+        Scan.cleanup_finished_scans(timedelta(minutes=1), log=True)
+
+        Scan.pause_non_ocr_conversions_on_scans_with_too_many_ocr_items()
+
+        time.sleep(10)
```
Comment (danchr) [inline scrapy-webscanner/process_manager.py]: Perhaps unrelated, but why ten seconds? That seems a rather long time to cease processing. Can't we do a blocking wait somewhere?

File: scrapy-webscanner/process_manager.py [hunk 3]
```diff
@@ -177,6 +177,106 @@ def main():
...
+                    # Check if process is still running
+                    os.kill(scan.pid, 0)
+                    logging.log(logging.DEBUG, 'Scan {} (with PID {}): OK'.format(scan.pk, scan.pid))
+                except OSError as ex:
+                    logging.log(logging.DEBUG, 'Scan {} (with PID {}): FAILED ({})'.format(scan.pk, scan.pid, str(ex)))
+                    scan.set_scan_status_failed(
+                        "SCAN FAILED: Process died with pid {}".format(scan.pid))
+            logging.log(logging.DEBUG, "Checked {} scans.".format(len(running_scans)))
+    except (DatabaseError, IntegrityError) as ex:
+        logging.log(logging.ERROR, 'Error occured while trying to kill process {}'.format(scan.pid))
```
Comment (magenta-af) [inline scrapy-webscanner/process_manager.py]: I just noticed (while working on #14) that `scan` isn't necessarily set here...

```
Traceback (most recent call last):
  File "/vagrant/scrapy-webscanner/process_manager.py", line 310, in <module>
    main()
  File "/vagrant/scrapy-webscanner/process_manager.py", line 193, in main
    check_running_scanjobs()
  File "/vagrant/scrapy-webscanner/process_manager.py", line 224, in check_running_scanjobs
    logging.log(logging.ERROR, 'Error occured while trying to kill process {}'.format(scan.pid))
UnboundLocalError: local variable 'scan' referenced before assignment
```

Comment (dals83) [reply scrapy-webscanner/process_manager.py]: No that off cause makes sense. If a db exception occurs the object collected from db is not present. The log message should be changed to not use scan.pid.

**Pull request merged**